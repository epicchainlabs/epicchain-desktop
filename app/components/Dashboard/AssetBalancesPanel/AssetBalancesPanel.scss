@import '../../../styles/animations';

.assetBalancesPanel {
  .centeredContent {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    padding: 0;
  }

  .header {
    display: flex;
    align-items: center;
    justify-content: space-between;

    .refresh {
      cursor: pointer;

      &,
      & path {
        color: #9599a2;
        fill: currentColor;
      }

      &.loading {
        animation: spin 2s linear infinite;
      }
    }
  }

  .assets {
    margin: 0 24px 10px 24px;
    display: flex;
    align-items: flex-start;
    justify-content: space-around;

    .asset {
      text-align: center;

      .quantity {
        font-weight: 400;
        margin: -8px 0 16px 0;
        font-family: var(--font-gotham-light);
        font-size: 40px;
        color: var(--base-text);
      }

      .value {
        color: #969aa0;
        font-size: 14px;
      }

      .change {
        margin-left: 10px;
        color: white;
        font-size: 14px;
        font-weight: bold;
        border-radius: 3px;
        padding: 1px 4px;

        &.increase {
          background-color: #5abf6b;
        }

        &.decrease {
          background-color: #ee6d66;
        }
      }
    }
  }

  .walletTotal {
    font-family: var(--font-gotham-medium);
    font-size: 24px;
    color: var(--base-text);
    text-align: center;
    line-height: 32px;
  }

  .totalValue {
    padding: 24px;
    display: flex;
    background: var(--dashboard-asset-panel-total-background);
    justify-content: space-between;
    align-items: center;
    font-size: 24px;
    font-weight: 500;
    padding: 12px 24px;
  }

  .label {
    font-size: 24px;
    font-weight: 400;
    text-transform: uppercase;
    font-family: var(--font-gotham-light);
    font-size: 24px;
    color: var(--base-text);
  }

  .assetName {
    opacity: 0.5;
    font-family: var(--font-gotham-bold);
    font-size: 13px;
    color: var(--dashboard-asset-panel-asset-name);
    text-align: center;
  }

  .claim {
    margin: 0 24px 12px;
    width: 350px;
    align-self: center;

    .claimButton {
      width: 100%;
      margin-bottom: 12px;
    }
  }
}
