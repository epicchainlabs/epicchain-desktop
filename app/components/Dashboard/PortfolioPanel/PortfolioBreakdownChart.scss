.portfolioBreakdownChart {
  tspan {
    fill: var(--dashboard-total-wallet-value-total-text);
  }
}
