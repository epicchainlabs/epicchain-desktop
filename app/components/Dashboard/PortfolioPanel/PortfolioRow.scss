.portfolioRow {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  padding: 10px 10px 10px 40px;
  align-items: center;
  width: 100%;

  .color,
  .symbol,
  .balance,
  .value,
  .percent {
    font-size: 13px;
    font-weight: 400;
    color: var(--base-text);
    overflow: hidden;
    text-overflow: ellipsis;
    width: 20%;
  }

  .color {
    flex: 0 0 auto;
    width: 15px;
    height: 15px;
    border-radius: 50%;
  }

  .percent {
    color: #969aa0;
    font-size: 13px;
    font-weight: 600;
    text-transform: uppercase;
  }
}

.oddColor {
  background-color: var(--dashboard-total-wallet-value-odd-row);
}
