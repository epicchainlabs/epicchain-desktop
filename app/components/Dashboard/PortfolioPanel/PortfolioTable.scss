.portfolioTable {
  overflow: hidden;
  align-self: center;
  // TODO
}
