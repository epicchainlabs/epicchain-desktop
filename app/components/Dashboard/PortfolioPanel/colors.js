export default ['#6AD8FF', '#3fd0ae', '#EE6D66', '#ee85c0', '#5ABF6B']
