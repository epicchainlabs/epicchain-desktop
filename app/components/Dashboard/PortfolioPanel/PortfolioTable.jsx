// @flow
import React from 'react'
import classNames from 'classnames'
import { values, times } from 'lodash-es'

import COLORS from './colors'
import PortfolioRow from './PortfolioRow'

import styles from './PortfolioTable.scss'

type BalanceType = {
  symbol: SymbolType,
  balance: string,
  value: number,
  percent: number,
}

type Props = {
  className?: string,
  balances: { [key: SymbolType]: BalanceType },
  currency: string,
}

export default class PortfolioTable extends React.Component<Props> {
  render = (): React$Node => {
    const { className, currency } = this.props
    const data = values(this.props.balances)

    return (
      <div className={classNames(styles.portfolioTable, className)}>
        {times(data.length, index => (
          <PortfolioRow
            {...data[index]}
            key={data[index].symbol}
            index={index}
            color={COLORS[index % COLORS.length]}
            currency={currency}
          />
        ))}
      </div>
    )
  }
}
