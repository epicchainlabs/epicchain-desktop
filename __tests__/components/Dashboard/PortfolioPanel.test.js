import React from 'react'
import nock from 'nock'
import { merge } from 'lodash-es'
import { mount, shallow } from 'enzyme'
import { progressValues } from 'spunky'
import { expect } from 'chai'

import { createStore, provideStore, provideState } from '../../testHelpers'
import AssetBalancesPanel from '../../../app/components/Dashboard/PortfolioPanel'
import {
  DEFAULT_CURRENCY_CODE,
  MAIN_NETWORK_ID,
  ASSETS,
  DEFAULT_LANGUAGE,
} from '../../../app/core/constants'
import PortfolioPanel from '../../../app/components/Dashboard/PortfolioPanel'
import PortfolioRow from '../../../app/components/Dashboard/PortfolioPanel/PortfolioRow'
import PortfolioTable from '../../../app/components/Dashboard/PortfolioPanel/PortfolioTable'
import PortfolioBreakdownChart from '../../../app/components/Dashboard/PortfolioPanel/PortfolioBreakdownChart'
import IntlWrapper from '../../../app/components/Root/IntlWrapper'

const { LOADED } = progressValues

const initialState = {
  spunky: {
    network: {
      batch: false,
      progress: LOADED,
      data: MAIN_NETWORK_ID,
      loadedCount: 1,
    },
    auth: {
      batch: false,
      progress: LOADED,
      data: {
        address: 'ANqUrhv99rwCiFTL6N1An9NH5UVkPYxTuw',
      },
      loadedCount: 1,
    },
    settings: {
      batch: false,
      progress: LOADED,
      data: {
        currency: DEFAULT_CURRENCY_CODE,
        language: DEFAULT_LANGUAGE,
        tokens: [],
      },
      loadedCount: 1,
    },
    prices: {
      batch: false,
      progress: LOADED,
      data: {
        NEO: 25.48,
        GAS: 18.1,
      },
      loadedCount: 1,
    },
    priceHistory: {
      batch: false,
      progress: LOADED,
      data: {
        [ASSETS.NEO]: [
          {
            close: 5,
          },
          ...new Array(29).fill({}),
          {
            close: 10,
          },
        ],
        [ASSETS.GAS]: [
          {
            close: 10,
          },
          ...new Array(29).fill({}),
          {
            close: 5,
          },
        ],
      },
      loadedCount: 1,
    },
    balances: {
      batch: false,
      progress: LOADED,
      data: {
        NEO: '100001',
        GAS: '1000.0001601',
      },
      loadedCount: 1,
    },
    claims: {
      batch: false,
      progress: LOADED,
      data: {
        total: '0.5',
      },
      loadedCount: 1,
    },
  },
  claim: {
    claimRequest: false,
    disableClaimButton: false,
  },
}

describe('PortfolioPanel', () => {
  beforeEach(() => {
    const response = [
      { symbol: 'NEO', price_usd: 24.5 },
      { symbol: 'GAS', price_usd: 18.2 },
    ]
    nock('https://api.coinmarketcap.com')
      .get('/v1/ticker/')
      .query({ limit: 0, convert: 'USD' })
      .reply(200, response, { 'Access-Control-Allow-Origin': '*' })
  })

  test('renders without crashing', () => {
    const store = createStore(initialState)
    const wrapper = shallow(<PortfolioPanel store={store} />)
    expect(wrapper).toMatchSnapshot()
  })

  test.only('renders <PortfolioBreakDownChart/>', () => {
    const store = createStore(initialState)
    const wrapper = mount(
      provideStore(
        <IntlWrapper>
          <PortfolioPanel />{' '}
        </IntlWrapper>,
        store,
      ),
    )
    expect(wrapper.find(PortfolioBreakdownChart)).to.have.length(1)
  })

  test.only('renders 2 <PortfolioRow/> for two assets saturated from store', () => {
    const store = createStore(initialState)
    const wrapper = mount(
      provideStore(
        <IntlWrapper>
          <PortfolioPanel />
        </IntlWrapper>,
        store,
      ),
    )
    expect(wrapper.find(PortfolioRow)).to.have.length(2)
  })

  test('account data refreshes when refresh button is clicked', () => {
    const store = createStore(initialState)
    const wrapper = mount(
      provideStore(
        <IntlWrapper>
          <PortfolioPanel />
        </IntlWrapper>,
        store,
      ),
    )

    wrapper
      .find('#refresh')
      .hostNodes()
      .simulate('click')

    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        type: 'balances/ACTION/CALL',
        meta: expect.objectContaining({ id: 'balances' }),
      }),
    )
  })
})
