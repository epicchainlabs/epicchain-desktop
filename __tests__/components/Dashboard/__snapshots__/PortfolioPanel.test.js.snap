// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioPanel renders without crashing 1`] = `
<withActions(Connect(withProgress(withProgressComponents(Connect(withData(Connect(withData(Connect(withData(withProps(withProps(PortfolioPanel))))))))))))
  onRetry={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
